clang/lib/Sema/SemaAPINotes.cpp

```diff
@@ -1044,11 +1044,13 @@ void Sema::ProcessAPINotes(Decl *D) {
 
   if (auto TagContext = dyn_cast<TagDecl>(D->getDeclContext())) {
     if (auto CXXMethod = dyn_cast<CXXMethodDecl>(D)) {
-      for (auto Reader : APINotes.findAPINotes(D->getLocation())) {
-        if (auto Context = UnwindTagContext(TagContext, APINotes)) {
-          auto Info =
-              Reader->lookupCXXMethod(Context->id, CXXMethod->getName());
-          ProcessVersionedAPINotes(*this, CXXMethod, Info);
+      if (CXXMethod->getIdentifier()) {
```

Comment:
I think that this is better spelt as `isa<NamedDecl>(CXXMethod)`. The `IdentifierInfo` is not preserved/used.

Reply:
`clang::CXXMethodDecl`s are always NamedDecls, right? But not all of them actually have a name that is convertible to a string.

Reply:
Right, it would always succeed. The names that you are referring to are all the operator overloads right? In that case, `isOverloadedOperator` might be a good option for the check (in the negative).

Reply:
Operators are one of the cases where `->getName()` would fail with an assertion. Other cases are destructors, constructors.

Reply:
`isa<CXXConstructorDecl>` and `isa<CXXDestructorDecl>` would identify those. The important thing is that it expresses concretely what you are trying to avoid.

Reply:
Alright, I replaced this with an explicit check for operators, constructors and destructors.